CDP4JsonFileDal/JsonFileDalUtils.cs

```diff
@@ -140,16 +140,33 @@ internal static void AddIterationSetup(IterationSetup iterationSetup, ref HashSe
         /// <param name="engineeringModelSetup">
         /// The <see cref="EngineeringModelSetup"/> that contains the active <see cref="DomainOfExpertise"/> instances.
         /// </param>
-        /// <param name="domainOfExpertises">
+        /// <param name="domainsOfExpertise">
         /// The target <see cref="HashSet{DomainOfExpertise}"/>
         /// </param>
-        internal static void AddDomainsOfExpertise(EngineeringModelSetup engineeringModelSetup, ref HashSet<DomainOfExpertise> domainOfExpertises)
+        internal static void AddDomainsOfExpertise(EngineeringModelSetup engineeringModelSetup, ref HashSet<DomainOfExpertise> domainsOfExpertise)
         {
             foreach (var domainOfExpertise in engineeringModelSetup.ActiveDomain)
             {
-                if (!domainOfExpertises.Contains(domainOfExpertise))
+                if (!domainsOfExpertise.Contains(domainOfExpertise))
+                {
+                    domainsOfExpertise.Add(domainOfExpertise);
+                }
+            }
+
+            foreach (var participant in engineeringModelSetup.Participant)
+            {
+                foreach (var domainOfExpertise in participant.Domain)
                 {
-                    domainOfExpertises.Add(domainOfExpertise);
+                    if (!domainsOfExpertise.Contains(domainOfExpertise))
+                    {
+                        domainsOfExpertise.Add(domainOfExpertise);
+                    }
+
+                    if (participant.Person.DefaultDomain != null &&
```

Comment:
This `if` should be outside the `domainOfExpertise` `foreach`.

Reply:
Correct, copy/paste error.